localdev/airflow/docker-compose.yaml

```diff
@@ -0,0 +1,309 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
This seems like a lot of tech debt to inherit.  Maybe we should just look into https://github.com/armadaproject/armada/issues/2255 and see how we can create an airflow instance via helm.  

I really don't think we want to support a third-party compose chart for airflow.  I'd rather use an official helm chart.

Reply:
I can look at using a helm chart instead.

Reply:
we already have an airflow service in the root docker-compose.yaml -- why not use that?

Reply:
The official airflow docker-compose.yaml gets us a better dag executor out of the box, as well as running the triggerer process (needed by deferrable operators) by default. It also gets us each airflow executable living in its own container instead of all of them being executed by the same container. 

You could argue we can adapt this script to do it as well, but the official docker-compose.yaml seemed easier to configure and work with.